iroh/src/endpoint/quic.rs

```diff
@@ -0,0 +1,491 @@
+//! Exporting and encapsulating structs from quinn
+//!
+//! Co-locates all iroh-quinn exports.
+//!
+//! There are some structs that we use in particular ways, where we would like
+//! to limit or expand how those structs are used in iroh. By encapsulating them
+//! we can ensure the functionality needed to make iroh work.
+
+use std::{sync::Arc, time::Duration};
+
+use n0_error::{e, stack_error};
+pub(super) use quinn::TransportConfig;
+// Missing still: SendDatagram and ConnectionClose::frame_type's Type.
+pub use quinn::{
+    AcceptBi, AcceptUni, AckFrequencyConfig, ApplicationClose, Chunk, ClosedStream,
+    ConnectionClose, ConnectionError, ConnectionStats, MtuDiscoveryConfig, OpenBi, OpenUni,
+    PathStats, ReadDatagram, ReadError, ReadExactError, ReadToEndError, RecvStream, ResetError,
+    RetryError, SendDatagramError, SendStream, ServerConfig, StoppedError, StreamId, VarInt,
+    VarIntBoundsExceeded, WeakConnectionHandle, WriteError,
+};
+pub use quinn_proto::{
+    FrameStats, IdleTimeout, TransportError, TransportErrorCode, UdpStats, Written,
+    congestion::{Controller, ControllerFactory},
+    crypto::{
+        AeadKey, CryptoError, ExportKeyingMaterialError, HandshakeTokenKey,
+        ServerConfig as CryptoServerConfig, UnsupportedVersion,
+    },
+};
```

Comment:
I never know if quinn should be re-exporting all of quinn-proto types that need to be pub as well. I think that would make sense. But certainly for multipath we haven't tackled that yet, and they just end up being wherever they are.

If later we move these types to be re-exported by quinn, would that be a breaking change for iroh? I think not, so it's probably fine. But writing this down here to get sanity checked by someone else.

Reply:
shouldn't be a breaking change, since `use` just makes an alias to an underlying type, it doesn't re-create it in the mod it gets `use`-d in, so it has the same underlying source.